server/index.js

```diff
@@ -1,104 +1,163 @@
-const pages = requireWithFallback('./routes/pages') // AdHoc: Changed to requireWithFallback
-const categories = requireWithFallback('./routes/categories') // AdHoc: Changed to requireWithFallback
-const playlists = requireWithFallback('./routes/playlists') // AdHoc: Changed to requireWithFallback
-const readingHistory = requireWithFallback('./routes/readingHistory') // AdHoc: Changed to requireWithFallback
-const redirects = requireWithFallback('./routes/redirects') // AdHoc: Changed to requireWithFallback
-const errorPages = requireWithFallback('./routes/errors') // AdHoc: Changed to requireWithFallback
-
-const userAuth = requireWithFallback('userAuth')
-const customCsp = requireWithFallback('csp')
-
-const app = express()
-// AdHoc: Added Body Parser to support posts
-app.use(express.json())
-
-const {preload, postload} = allMiddleware
-
-// The trust proxy flag tells the app to use https for links
-// and redirect urls if it sees indications that the request
-// passed through a proxy and was originally sent using https
-if ((process.env.TRUST_PROXY || '').toUpperCase() === 'TRUE') {
-  app.enable('trust proxy')
-}
+const log = require('./logger');
 
-app.set('view engine', 'ejs')
-app.set('views', [path.join(__dirname, '../custom/layouts'), path.join(__dirname, '../layouts')])
+(async () => {
+  const gcpMetadata = require('gcp-metadata');
+  const { SecretManagerServiceClient } = require('@google-cloud/secret-manager');
 
-app.get('/healthcheck', (req, res) => {
-  res.send('OK')
-})
+  // GCP Metadata is only available to the deployed instance.
+  // Running locally, you would still use the .env file.
+  const isDeployedInstance = await gcpMetadata.isAvailable();
 
-app.use(csp({directives: customCsp}))
-app.use(userAuth)
+  if (isDeployedInstance) {
+    log.info('App Engine deployed instance detected...');
 
-preload.forEach((middleware) => app.use(middleware))
+    // We're on GCP here.  Load up environment variables from 
+    // Secrets Manager
+    const projectId = await gcpMetadata.project('project-id');
 
-app.use(userInfo)
+    log.info(`Loading environment variables for project: ${projectId}`);
 
-// serve all files in the public folder
-app.use('/assets', express.static(path.join(__dirname, '../public')))
+    const base64Vars = ['GOOGLE_APPLICATION_JSON'];
 
-// strip trailing slashes from URLs
-app.get(/(.+)\/$/, (req, res, next) => {
-  res.redirect(req.params[0])
-})
+    const client = new SecretManagerServiceClient();
 
-app.get('/view-on-site/:docId', (req, res, next) => {
-  const {docId} = req.params
-  const doc = getMeta(docId)
+    const parent = `projects/${projectId}`;
 
-  if (!doc) return next(Error('Not found'))
+    const [secrets] = await client.listSecrets({
+      parent: parent,
+    });
 
-  res.redirect(doc.path)
-})
+    for (let secret of secrets) {
+      const path = secret.name.split('/');
+      const envVar = path[path.length > 0 ? path.length - 1 : 0];
```

Comment:
```suggestion
      const envVar = path[path.length - 1];
```

if there are no `/` present, `path.length` will still be `1`